### PR TITLE
Add note about using whitespace control for macros

### LINKED ISSUE #4910: Add whitespace control warning in generate_schema_name macro mentions
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Per [slack thread](https://dbt-labs.slack.com/archives/C019QLR6EJE/p1707879709105339), a user reported that we should include mentions of whitespace control in generate_schema_macro examples. 

A customer overriding the [generate_schema_name](https://docs.getdbt.com/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name) macro did not include the [whitespace control](https://docs.getdbt.com/faqs/Jinja/jinja-whitespace) dashes, which had some effects on their project.  Perhaps we could add that as a warning in the docs anywhere we mention overriding macros. 

### What part(s) of the page would you like to see updated?

https://docs.getdbt.com/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
Resolves #4910

this pr adds info about needing to add whitespace (dashes or minuses) in macro to remove whitespace adn avoid issues

https://github.com/dbt-labs/docs.getdbt.com/issues/4910

### CODE REVIEW

File: website/snippets/_whitespace-control.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+<expandable alt_header="📌 Tidy your macros with whitespace control">
+
+When you're changing macros, like the `generate_schema_name`, in your project &mdash; you might notice extra white space in your code in the `target/compiled` folder.
```
Comment (nghi-ly) [inline website/snippets/_whitespace-control.md]:
```suggestion
When you're changing macros &mdash; like the `generate_schema_name` &mdash; in your project, you might notice extra whitespace in your code in the `target/compiled` folder. 
```

File: website/snippets/_whitespace-control.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+<expandable alt_header="📌 Tidy your macros with whitespace control">
+
+When you're changing macros, like the `generate_schema_name`, in your project &mdash; you might notice extra white space in your code in the `target/compiled` folder. 
+
+You can use Jinja's [whitespace control](/faqs/Jinja/jinja-whitespace) using a minus sign `(-)` to tidy that up and remove unwanted spaces/lines. For example, use `{{- ... -}}` or `{%- ... %}` around your macro definitions (such as `{%- macro generate_schema_name(...) -%} ... {%- endmacro -%}`).
```
Comment (nghi-ly) [inline website/snippets/_whitespace-control.md]: sugg: the `... can use ... using a` construction was slightly awkward to read.  

```suggestion
You can remove unwanted spaces and lines with Jinja's [whitespace control](/faqs/Jinja/jinja-whitespace) by using a minus sign. For example, use `{{- ... -}}` or `{%- ... %}` around your macro definitions (such as `{%- macro generate_schema_name(...) -%} ... {%- endmacro -%}`).
```

File: website/snippets/_whitespace-control.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+<expandable alt_header="📌 Tidy your macros with whitespace control">
```
Comment (nghi-ly) [inline website/snippets/_whitespace-control.md]: suggest removing the emoji. the red sorta clashes with the orange, at least to me. wdyt?

Screenshot 2024-02-16 at 9 05 27 AM

File: website/docs/docs/build/custom-schemas.md [hunk 4]
```diff
@@ -92,8 +96,7 @@ To customize this macro, copy the example code in the section [How does dbt gene
 
 Be careful. dbt will ignore any custom `generate_schema_name` macros included in installed packages.
 
-<details>
-<summary>❗️ Warning: Don't replace <code>default_schema</code> in the macro.</summary>
+<expandable alt_header="❗️ Warning: Don't replace `default_schema` in the macro">
```
Comment (nghi-ly) [inline website/docs/docs/build/custom-schemas.md]: same thing here re: the emoji

File: website/docs/docs/build/custom-schemas.md [hunk 5]
```diff
@@ -84,6 +84,10 @@ The following code represents the default macro's logic:
 {%- endmacro %}
 ```
 
+import WhitespaceControl from '/snippets/_whitespace-control.md';
+
+<WhitespaceControl/>
```
Comment (nghi-ly) [inline website/docs/docs/build/custom-schemas.md:90]: suggest moving this to line 67 above. it feels a little lost here between a code block and an H2 heading.

check the other instances and their placements as well.

**Review by nghi-ly — COMMENTED** [review]
looks great! left some comments around placement of expandable and emoji

File: website/snippets/_whitespace-control.md (hunk 3 above)
Comment (mirnawong1) [reply website/snippets/_whitespace-control.md]: thanks Ly! i added an emoji bc i thought it blended too much with text and didn't want users to miss this as it's a good tip. i also think it reads better after the code example and not in the middle of the config explanation. 

i don't think expendables works well in this use case so have updated it to details so it stands out a bit more. 
Screenshot 2024-02-19 at 10 43 36
Screenshot 2024-02-19 at 10 53 19

**Comment by mirnawong1** [comment]
thank you so much @nghi-ly! took some of your suggs but also thinking of the placement/flow of it so have changed it a little too. can you rereview? thank you!

**Comment by mirnawong1** [comment]
thanks @ChenyuLInx and @nghi-ly - ready for re-review for yall! auto merge enabled

File: website/snippets/_whitespace-control.md (hunk 3 above)
Comment (nghi-ly) [reply website/snippets/_whitespace-control.md]: nice!!

**Review by nghi-ly — APPROVED** [review]
:shipit:

**Pull request merged**